packages/cojson/src/sync.ts

```diff
@@ -482,28 +485,38 @@ export class SyncManager {
       )) {
         const dependencyCoValue = this.local.getCoValue(dependency);
 
-        if (!dependencyCoValue.verified) {
+        if (!dependencyCoValue.hasVerifiedContent()) {
           coValue.markMissingDependency(dependency);
 
-          if (!dependencyCoValue.verified) {
-            const peers = this.getServerAndStoragePeers();
+          const peers = this.getServerAndStoragePeers();
 
-            // if the peer that sent the content is a client, we add it to the list of peers
-            // to also ask them for the dependency
-            if (peer?.role === "client") {
-              peers.push(peer);
-            }
-
-            dependencyCoValue.load(peers);
+          // if the peer that sent the content is a client, we add it to the list of peers
+          // to also ask them for the dependency
+          if (peer?.role === "client") {
+            peers.push(peer);
           }
+
+          dependencyCoValue.load(peers);
         }
       }
 
       peer?.updateHeader(msg.id, true);
-      coValue.provideHeader(msg.header, peer?.id ?? "storage");
+      coValue.provideHeader(
+        msg.header,
+        peer?.id ?? "storage",
+        msg.streamingTarget,
+      );
+
+      if (msg.streamingTarget) {
+        peer?.combineWith(msg.id, {
+          id: msg.id,
+          header: true,
+          sessions: msg.streamingTarget,
+        });
+      }
```

Comment:
Here we track how much content the peer has, and since we know that the other peer is going to stream content to us, we track the streamingTarget here.